File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/secret/impl/PinPatternSecretPrompter.java
package at.tugraz.tc.cyfile.secret.impl;

import android.content.Context;
import android.content.Intent;

import at.tugraz.tc.cyfile.secret.SecretPrompter;
import at.tugraz.tc.cyfile.ui.PatternLockActivity;

import static com.blankj.utilcode.util.ActivityUtils.startActivity;

public class PinPatternSecretPrompter implements SecretPrompter {

    private Context context;

    public PinPatternSecretPrompter(Context context) {
        this.context = context;
    }

    @Override
    public void promptSecret() {
        Intent intent = new Intent(context, PatternLockActivity.class);
        startActivity(intent);
    }


}

File: CyFile-Android/app/src/main/kotlin/at/tugraz/tc/cyfile/secret/impl/PinPatternSecretPrompter.kt
package at.tugraz.tc.cyfile.secret.impl

import android.content.Context
import android.content.Intent

import at.tugraz.tc.cyfile.secret.SecretPrompter
import at.tugraz.tc.cyfile.ui.PatternLockActivity

import com.blankj.utilcode.util.ActivityUtils.startActivity

class PinPatternSecretPrompter(private val context: Context) : SecretPrompter {

    override fun promptSecret() {
        val intent = Intent(context, PatternLockActivity::class.java)
        startActivity(intent)
    }


}
